coalals/interface.py

```diff
@@ -0,0 +1,60 @@
+import sys
+from os import chdir
+from json import dumps
+from io import StringIO
+from coalib import coala
+from contextlib import redirect_stdout
+from concurrent.futures import ProcessPoolExecutor
+from .utils.log import configure_logger, reset_logger
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class coala_wrapper:
+
+    def __init__(self, proxy_map, max_jobs=1, max_workers=1):
+        self._proxy_map = proxy_map
+        self._process_pool = ProcessPoolExecutor(max_workers=max_workers)
+
+        self._jobs = []
+        self._max_jobs = max_jobs
+
+    def refresh_jobs(self):
+        self._jobs = list(filter(lambda job: job.done(), self._jobs))
+
+    @staticmethod
+    def _run_coala():
+        stream, retval = StringIO(), -1
+        with redirect_stdout(stream):
+            return (stream, coala.main())
+
+    @staticmethod
+    def _get_op_from_coala(stream, retval):
+        output = None
+        if retval == 1:
+            output = stream.getvalue()
+            if output:
+                logger.debug('Output: %s', output)
+            else:
+                logger.debug('No results for the file')
+        elif retval == 0:
+            logger.debug('No issues found')
+        else:
+            logger.debug('Exited with: %s', retval)
+
+        return output or dumps({'results': {}})
+
+    def analyse_file(self, file_proxy):
+        sys.argv = ['', '--json', '--find-config',
+                    '--limit-files', file_proxy.filename]
+
+        workspace = file_proxy.workspace
+        if workspace is None:
+            workspace = '.'
+
+        # change workspace
```

Comment:
The comment does not have additional information, maybe we could remove it or rewrite it.